src/main.rs

```diff
@@ -166,15 +173,28 @@ fn inner_main(language: &mut Option<Language>) -> errors::Result<()> {
 
    log::debug!("init done! starting event loop");
 
+   let mut last_window_size: PhysicalSize<u32> = PhysicalSize::default();
+   let mut last_window_position: PhysicalPosition<i32> = PhysicalPosition::default();
    event_loop.run(move |event, _, control_flow| {
       *control_flow = ControlFlow::Poll;
 
       match event {
          Event::WindowEvent { event, .. } => {
-            if let WindowEvent::CloseRequested = event {
-               *control_flow = ControlFlow::Exit;
-            } else {
-               input.process_event(&event);
+            match event {
+               // Ignore resize event if window is maximized, and move event if position is lower than 0,
+               // because it isn't what we want, when saving window's size and position to config file.
+               WindowEvent::Resized(new_size) if !ui.window().is_maximized() => {
+                  last_window_size = new_size;
+               }
+               WindowEvent::Moved(new_position) if new_position.x >= 0 && new_position.y >= 0 => {
+                  last_window_position = new_position;
+               }
```

Comment:
Note that moving the window outside the screen (crossing the left border of your display) is a perfectly legit thing to do on stacking window managers, so positions < 0 are a normal thing. Unless you meant to clamp them between `[0; display_width - window_width)`, but I don't really see a reason to do that.

Reply:
Sorry, an inaccuracy on my part. This if is to check if the window was moved not because of maximization. On Windows after maximization the window position (in my case) changes to (-8, -8). At first it was just checking if the window was maximized (like WindowEvent::Resized), but that still let it through.

Reply:
I think we can just keep that; if the window is maximized it shouldn't matter anyways *at least in theory*. Once the window is maximized its position doesn't matter until it's unmaximized. I can see this becoming a problem if the user clicks the unmaximize button, but I don't think we can do anything reasonable in that case. I'd suggest reporting this strange behavior to winit.

I tested on winit's X11 and Wayland backends. X11 doesn't fire the event upon maximization, and Wayland never fires Moved events at all.

Reply:
If this solution suits you, then this conversation can be resolved and everything is, to me, done.

Reply:
I guess yeah, it can work. Restricting the position like this shouldn't hurt anything because it's not very common that people put windows partially outside their display. At least I don't.